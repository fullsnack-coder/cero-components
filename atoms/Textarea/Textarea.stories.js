import { Textarea, styles } from '.'

import { getTemplate } from '@glrodasz/storybook-tools-helpers'

const Template = getTemplate(Textarea, styles)

export default {
  title: 'Atoms/Textarea',
  component: Textarea,
  args: {
    placeholder:
      'Country speed seat newfound ah stained Gundabad stinking runt.',
  },
}

export const Default = Template.bind({})
