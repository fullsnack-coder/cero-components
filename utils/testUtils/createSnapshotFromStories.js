import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
