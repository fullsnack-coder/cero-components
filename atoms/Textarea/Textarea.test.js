import * as stories from './Textarea.stories'
import createSnapshotFromStories from '../../utils/testUtils/createSnapshotFromStories'

describe('[ atoms / Textarea ]', () => {
  describe('#Textarea', () => createSnapshotFromStories('Textarea', stories))
})
