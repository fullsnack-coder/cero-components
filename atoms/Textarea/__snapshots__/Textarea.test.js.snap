// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ atoms / Textarea ] #Textarea when 'Textarea' Default is mounted should render 1`] = `
<DocumentFragment>
  <textarea
    class="textarea"
    placeholder="Country speed seat newfound ah stained Gundabad stinking runt."
    rows="5"
  />
</DocumentFragment>
`;
